File: cos/cos.olympus/src/main/java/cos/olympus/game/events/Damage.java
package cos.olympus.game.events;

import cos.olympus.game.Creature;

public record Damage(
        int id,
        int tick,
        Creature victim,
        Spell spell,
        int amount,
        boolean crit
) {

    public cos.ops.Damage toUserOp(int userId) {
        return new cos.ops.Damage(id, tick, userId, victim.id(), amount, spell.id(), crit);
    }

    @Override public String toString() {
        return "Damage{" +
                "id=" + id +
                ", tick=" + tick +
                ", victim=" + victim.id() +
                ", spell=" + spell.id() +
                ", amount=" + amount +
                '}';
    }
}

File: cos/cos.olympus/src/main/kotlin/cos/olympus/game/actions/Damage.kt
package cos.olympus.game.actions

import cos.olympus.Tsm
import cos.olympus.game.Creature

data class Damage(
    override val x: Int,
    override val y: Int,
    override val time: Tsm,
    val victim: Creature,
    val culprit: Creature,
    var amount: Int = 0,
    val spellId: Long = 0
) : Action
